### ISSUE TITLE
[typescript-urql-svelte] New Codegen Plugin Request

### ISSUE THREAD

**Issue by trevyn** [dotansimha/graphql-code-generator#5400]
The existing `typescript-urql` plugin is specific to React, and [Svelte](https://svelte.dev/) is awesome. I have [`@urql/svelte`](https://formidable.com/open-source/urql/docs/api/svelte/) up and running, and would love to see a `typescript-urql-svelte` plugin for it.

As of today, the Svelte [repo](https://github.com/sveltejs/svelte) has 41.4k stars, so it should be around a while. :)

**Comment by ardatan** [comment]
@trevyn That sounds great! Would you start with a draft PR to help us? We don't know much about Svelte.

**Comment by trevyn** [comment]
Sure, I'll take a look. For now, [here's my quick-and-dirty `@urql/svelte` impl](https://github.com/trevyn/turbo/blob/375e7a42fb707970dc462ce25272e19968c64e8b/frontend/App.svelte). I honestly am not sure about the TypeScript part yet, but I'll keep this thread updated and submit a draft PR if I figure out a sensible plugin. :)

**Comment by trevyn** [comment]
Ok, I did a reasonable draft of what the codegen output could look like, and verified it works:

[`App.svelte`](https://github.com/trevyn/turbo/blob/45fe80fa5cd05fd9d381aedc44e2588fbcdad623/frontend/App.svelte) (usage)
[`graphql_ops.svelte`](https://github.com/trevyn/turbo/blob/45fe80fa5cd05fd9d381aedc44e2588fbcdad623/frontend/graphql_ops.svelte) (would be codegen'd)

This is just in straight JS, I'll need to add TypeScript.

I also looked at the plugin code for `typescript-urql` to understand how to actually make the plugin. It looks OK, so I'll probably do that over the next few days. (Unless this is already enough, and you have someone that wants to take it on?)
